### ISSUE TITLE
Use more explicit keywords

### ISSUE THREAD

**Issue by jrbourbeau** [dask/dask#8331]
Our top-level `dask.compute`, `dask.persist`, and `dask.visualize` methods all have `**kwargs` in their signature. Certain keyword arguments are then `pop`ed off the `kwargs` dict 

https://github.com/dask/dask/blob/89d93a8264ce81c30207d83de96e3346a15365a4/dask/base.py#L551-L552

and the remaining `kwargs` are then forwarded elsewhere

https://github.com/dask/dask/blob/89d93a8264ce81c30207d83de96e3346a15365a4/dask/base.py#L564

It would be nice if the keywords which were `pop`ed off `kwargs` were instead explicitly marked in the method signature. For example, if `compute`s signature was changed from 

```python
def compute(*args, **kwargs):
```

to 

```python
def compute(*args, traverse=True, scheduler=None, optimize_graph=True, **kwargs):
```

This would be it easier to see what keyword arguments are supported by quickly inspecting these method's signature and would probably lead to better type checking integration. This is relatively low priority, but would be a nice quality of life improvement that, I think, should be relatively straightforward to implement.

**Comment by gjoseph92** [comment]
This would be nice to do in many, many places. I assume this is all holdover from Python 2 syntax. It both benefits users and developers, since tracking what happens to `kwargs` as they get passed down to different functions that are all popping things off is a pain. Maybe we should make a meta-issue tracking all the places to do this?

**Comment by jrbourbeau** [comment]
Sure, making this replacement in other places seems fine too. `compute`, `persist`, and `visualize` just stood out as particularly well-worn parts of our public API.  Below are all the times `kwargs.pop(` occurs throughout the codebase (according to `git grep "kwargs.pop("`). I've not validated that all these occurrences merit being replaced, but I suspect many of them could be. 

<details>
<summary>"kwargs.pop(" occurances:</summary>

```
dask/array/backends.py:        kwargs.pop("casting", None)
dask/array/backends.py:        kwargs.pop("order", None)
dask/array/core.py:    out = kwargs.pop("out", None)
dask/array/core.py:    dtype = kwargs.pop("enforce_dtype")
dask/array/core.py:    function = kwargs.pop("enforce_dtype_function")
dask/array/core.py:    subok = bool(kwargs.pop("subok", False))
dask/array/core.py:    chunks = kwargs.pop("chunks", True)
dask/array/creation.py:    chunks = kwargs.pop("chunks", "auto")
dask/array/creation.py:    like = kwargs.pop("like", None)
dask/array/creation.py:    dtype = kwargs.pop("dtype", None)
dask/array/creation.py:    indexing = kwargs.pop("indexing", "xy")
dask/array/creation.py:    sparse = bool(kwargs.pop("sparse", False))
dask/array/einsumfuncs.py:    subscripts = kwargs.pop("subscripts")
dask/array/einsumfuncs.py:    ncontract_inds = kwargs.pop("ncontract_inds")
dask/array/einsumfuncs.py:    dtype = kwargs.pop("kernel_dtype")
dask/array/einsumfuncs.py:    dtype = kwargs.pop("dtype", None)
dask/array/einsumfuncs.py:    optimize = kwargs.pop("optimize", False)
dask/array/einsumfuncs.py:    split_every = kwargs.pop("split_every", None)
dask/array/gufunc.py:        self.signature = kwargs.pop("signature", None)
dask/array/gufunc.py:        self.vectorize = kwargs.pop("vectorize", False)
dask/array/gufunc.py:        self.axes = kwargs.pop("axes", None)
dask/array/gufunc.py:        self.axis = kwargs.pop("axis", None)
dask/array/gufunc.py:        self.keepdims = kwargs.pop("keepdims", False)
dask/array/gufunc.py:        self.output_sizes = kwargs.pop("output_sizes", None)
dask/array/gufunc.py:        self.output_dtypes = kwargs.pop("output_dtypes", None)
dask/array/gufunc.py:        self.allow_rechunk = kwargs.pop("allow_rechunk", False)
dask/array/gufunc.py:        self.meta = kwargs.pop("meta", None)
dask/array/gufunc.py:            allow_rechunk=self.allow_rechunk or kwargs.pop("allow_rechunk", False),
dask/array/ma.py:    kwargs.pop("chunks", None)  # A Dask kwarg, not NumPy.
dask/array/ma.py:    dtype = kwargs.pop("masked_dtype", None)
dask/array/overlap.py:        drop_axis = kwargs.pop("drop_axis", None)
dask/array/routines.py:    axis = kwargs.pop("axis", None)
dask/array/tests/test_array_core.py:        cast = kwargs.pop("cast", "i8")
dask/array/tests/test_array_core.py:        assert kwargs.pop("foo") == "test kwarg"
dask/array/tests/test_cupy_core.py:        assert kwargs.pop("foo") == "test kwarg"
dask/array/tests/test_gufunc.py:        cast = kwargs.pop("cast", "i8")
dask/array/tiledb_io.py:        key = kwargs.pop("key", None)
dask/array/ufunc.py:            func = partial(self._ufunc.outer, dtype=kwargs.pop("dtype"))
dask/array/wrap.py:    name = kwargs.pop("name", None)
dask/array/wrap.py:    chunks = kwargs.pop("chunks", "auto")
dask/array/wrap.py:    dtype = kwargs.pop("dtype", None)
dask/array/wrap.py:        shape = kwargs.pop("shape")
dask/array/wrap.py:    func_like = kwargs.pop("func_like", None)
dask/base.py:    traverse = kwargs.pop("traverse", True)
dask/base.py:    traverse = kwargs.pop("traverse", True)
dask/base.py:    optimize_graph = kwargs.pop("optimize_graph", True)
dask/base.py:        scheduler=kwargs.pop("scheduler", None),
dask/base.py:        get=kwargs.pop("get", None),
dask/base.py:    filename = kwargs.pop("filename", "mydask")
dask/base.py:    optimize_graph = kwargs.pop("optimize_graph", False)
dask/base.py:            cmap = kwargs.pop("cmap")
dask/base.py:        maxval = kwargs.pop("maxval", None)
dask/base.py:    traverse = kwargs.pop("traverse", True)
dask/base.py:    optimize_graph = kwargs.pop("optimize_graph", True)
dask/base.py:        scheduler=kwargs.pop("scheduler", None), collections=collections
dask/dataframe/accessor.py:            meta = kwargs.pop("meta")
dask/dataframe/backends.py:    ignore_order = kwargs.pop("ignore_order", False)
dask/dataframe/core.py:    meta = kwargs.pop("meta", no_default)
dask/dataframe/core.py:    out = kwargs.pop("out", None)
dask/dataframe/core.py:    transform_divisions = kwargs.pop("transform_divisions", True)
dask/dataframe/core.py:    with raise_on_meta_error(funcname(func), udf=kwargs.pop("udf", False)):
dask/dataframe/core.py:    name = kwargs.pop("token", None)
dask/dataframe/core.py:    parent_meta = kwargs.pop("parent_meta", None)
dask/dataframe/core.py:    func = kwargs.pop("_func")
dask/dataframe/core.py:    meta = kwargs.pop("_meta")
dask/dataframe/groupby.py:    func = kwargs.pop("chunk")
dask/dataframe/groupby.py:    columns = kwargs.pop("columns")
dask/dataframe/groupby.py:    name = kwargs.pop("name")
dask/dataframe/groupby.py:    funcs = kwargs.pop("funcs")
dask/dataframe/groupby.py:    levels = kwargs.pop("index_levels")
dask/dataframe/groupby.py:    levels = kwargs.pop("index_levels")
dask/dataframe/io/csv.py:            rest_kwargs.pop("skiprows", None)
dask/dataframe/io/csv.py:            rest_kwargs.pop("skipfooter", None)
dask/dataframe/io/csv.py:    head_kwargs.pop("skipfooter", None)
dask/dataframe/io/json.py:        name_function=kwargs.pop("name_function", None),
dask/dataframe/io/parquet/arrow.py:        partitioning = _dataset_kwargs.pop(
dask/dataframe/io/parquet/arrow.py:            partitioning = kwargs.pop("partitioning", None)
dask/dataframe/io/parquet/arrow.py:    ignore_metadata_file = kwargs.pop("ignore_metadata_file", False)
dask/dataframe/io/parquet/fastparquet.py:        parquet_file = kwargs.pop("parquet_file", None)
dask/dataframe/io/parquet/fastparquet.py:        object_encoding = kwargs.pop("object_encoding", "utf8")
dask/dataframe/io/parquet/fastparquet.py:        index_cols = kwargs.pop("index_cols", [])
dask/dataframe/io/tests/test_parquet.py:    oe = write_kwargs.pop("object_encoding", None)
dask/dataframe/multi.py:    empty_index_dtype = kwargs.pop("empty_index_dtype", None)
dask/dataframe/multi.py:    categorical_columns = kwargs.pop("categorical_columns", None)
dask/dataframe/rolling.py:        func_name = kwargs.pop("token")
dask/dataframe/rolling.py:        meta = kwargs.pop("meta")
dask/delayed.py:    pure = kwargs.pop("pure", None)
dask/delayed.py:        pure = kwargs.pop("pure", None)
dask/delayed.py:        name = kwargs.pop("dask_key_name", None)
dask/delayed.py:    dask_key_name = kwargs.pop("dask_key_name", None)
dask/delayed.py:    pure = kwargs.pop("pure", pure)
dask/diagnostics/profile_visualize.py:        filename = kwargs.pop("file_path")
dask/diagnostics/profile_visualize.py:        kwargs["width"] = kwargs.pop("plot_width")
dask/diagnostics/profile_visualize.py:        kwargs["height"] = kwargs.pop("plot_height")
dask/diagnostics/profile_visualize.py:        kwargs["width"] = kwargs.pop("plot_width")
dask/diagnostics/profile_visualize.py:        kwargs["height"] = kwargs.pop("plot_height")
dask/diagnostics/profile_visualize.py:        kwargs.pop("label_size")
dask/diagnostics/profile_visualize.py:        kwargs["width"] = kwargs.pop("plot_width")
dask/diagnostics/profile_visualize.py:        kwargs["height"] = kwargs.pop("plot_height")
dask/local.py:    kwargs.pop("num_workers", None)  # if num_workers present, remove it
dask/utils.py:    function = kwargs.pop("function")
dask/utils.py:    other = kwargs.pop("other")
docs/source/custom-collections.rst:        if kwargs.pop('optimize_graph', True):

```

</details>

**Comment by boazmohar** [comment]
I think i did most of these in #8354, the places I am not sure how to change as I don't know how to document the arguments:
1.[`token`](https://github.com/dask/dask/blob/main/dask/dataframe/core.py#L5848)
2.[`parent_meta`](https://github.com/dask/dask/blob/main/dask/dataframe/core.py#L5849)

<details>
  <summary>Remaining places and some remarks about them</summary>
 
 ```
 ./dask/dask/base.py:685:            cmap = kwargs.pop("cmap")	No default
./dask/dask/diagnostics/profile_visualize.py:187:        filename = kwargs.pop("file_path") file_path is deprecated
./dask/dask/diagnostics/profile_visualize.py:255:        kwargs["width"] = kwargs.pop("plot_width")  for backwards compatibility
./dask/dask/diagnostics/profile_visualize.py:257:        kwargs["height"] = kwargs.pop("plot_height")  for backwards compatibility
./dask/dask/diagnostics/profile_visualize.py:358:        kwargs["width"] = kwargs.pop("plot_width")  for backwards compatibility
./dask/dask/diagnostics/profile_visualize.py:360:        kwargs["height"] = kwargs.pop("plot_height")  for backwards compatibility
./dask/dask/diagnostics/profile_visualize.py:364:        kwargs.pop("label_size") Drop `label_size` to match `plot_cache` and `plot_tasks` kwargs
./dask/dask/diagnostics/profile_visualize.py:460:        kwargs["width"] = kwargs.pop("plot_width") for backwards compatibility
./dask/dask/diagnostics/profile_visualize.py:462:        kwargs["height"] = kwargs.pop("plot_height") for backwards compatibility
./dask/dask/array/backends.py:127:        kwargs.pop("casting", None) Value not used: cupy does not accept `order` or `casting` kwargs - ignore
./dask/dask/array/backends.py:128:        kwargs.pop("order", None) Value not used: cupy does not accept `order` or `casting` kwargs - ignore
./dask/dask/array/ufunc.py:168:            func = partial(self._ufunc.outer, dtype=kwargs.pop("dtype")), No default
./dask/dask/array/tests/test_cupy_core.py:250:        assert kwargs.pop("foo") == "test kwarg"  No default
./dask/dask/array/tests/test_gufunc.py:457:        cast = kwargs.pop("cast", "i8") in tests, might not be needed
./dask/dask/array/tests/test_array_core.py:1509:        cast = kwargs.pop("cast", "i8") in tests, might not be needed
./dask/dask/array/tests/test_array_core.py:1664:        assert kwargs.pop("foo") == "test kwarg"  No default
./dask/dask/array/einsumfuncs.py:12:    subscripts = kwargs.pop("subscripts") No default
./dask/dask/array/einsumfuncs.py:13:    ncontract_inds = kwargs.pop("ncontract_inds")  No default
./dask/dask/array/einsumfuncs.py:14:    dtype = kwargs.pop("kernel_dtype") No default
./dask/dask/array/wrap.py:21:    name = kwargs.pop("name", None) not a **kwargs in function
./dask/dask/array/wrap.py:22:    chunks = kwargs.pop("chunks", "auto") not a **kwargs in function
./dask/dask/array/wrap.py:24:    dtype = kwargs.pop("dtype", None) not a **kwargs in function
./dask/dask/array/wrap.py:51:        shape = kwargs.pop("shape") No default
./dask/dask/array/overlap.py:701:        drop_axis = kwargs.pop("drop_axis", None) should be passed as kwargs in some cases
./dask/dask/array/core.py:4510:    dtype = kwargs.pop("enforce_dtype") No default
./dask/dask/array/core.py:4511:    function = kwargs.pop("enforce_dtype_function")No default
./dask/dask/dataframe/backends.py:376:    ignore_order = kwargs.pop("ignore_order", False) raised an issue 
./dask/dask/dataframe/accessor.py:61:            meta = kwargs.pop("meta") No default
./dask/dask/dataframe/rolling.py:95:        func_name = kwargs.pop("token") No default
./dask/dask/dataframe/rolling.py:102:        meta = kwargs.pop("meta") No default
./dask/dask/dataframe/io/tests/test_parquet.py:977:    oe = write_kwargs.pop("object_encoding", None) in tests
./dask/dask/dataframe/io/parquet/arrow.py:776:        partitioning = _dataset_kwargs.pop( 	not kwargs
./dask/dask/dataframe/io/parquet/arrow.py:1482:            partitioning = kwargs.pop("partitioning", None)  Not sure if change would be more readable
./dask/dask/dataframe/io/parquet/arrow.py:1624:    ignore_metadata_file = kwargs.pop("ignore_metadata_file", False) Came from another variable not function def
./dask/dask/dataframe/io/parquet/fastparquet.py:939:        parquet_file = kwargs.pop("parquet_file", None) Not sure if change would be more readable
./dask/dask/dataframe/groupby.py:289:    func = kwargs.pop("chunk")  No default
./dask/dask/dataframe/groupby.py:290:    columns = kwargs.pop("columns")  No default
./dask/dask/dataframe/groupby.py:514:    name = kwargs.pop("name") No default
./dask/dask/dataframe/groupby.py:906:    funcs = kwargs.pop("funcs")  No default
./dask/dask/dataframe/groupby.py:2070:    levels = kwargs.pop("index_levels")  No default
./dask/dask/dataframe/groupby.py:2080:    levels = kwargs.pop("index_levels")  No default
./dask/dask/dataframe/core.py:5848:    name = kwargs.pop("token", None) **Not sure how to document**
./dask/dask/dataframe/core.py:5849:    parent_meta = kwargs.pop("parent_meta", None) **Not sure how to document**
./dask/dask/dataframe/core.py:5981:    func = kwargs.pop("_func") No deafult
./dask/dask/dataframe/core.py:5982:    meta = kwargs.pop("_meta") No deafult
./dask/dask/utils.py:1218:    function = kwargs.pop("function") No deafult
./dask/dask/utils.py:1219:    other = kwargs.pop("other") No deafult
./dask/dask/delayed.py:605:    dask_key_name = kwargs.pop("dask_key_name", None) # kwargs is not **kwargs in function declaration
./dask/dask/delayed.py:606:    pure = kwargs.pop("pure", pure) # kwargs is not **kwargs in function declaration
./dask/dask/local.py:552:    kwargs.pop("num_workers", None)  # if num_workers present, remove it
./dask/docs/source/custom-collections.rst:278:        if kwargs.pop('optimize_graph', True): Not sure changing docs here helps
```
</details>

**Comment by jsignell** [comment]
Yeah I think `token` and `parent_meta` are really meant for internal use and aren't considered to be user-facing so it's fine to leave them as is.

**Comment by ian-r-rose** [comment]
Closing as fixed by #8354, thanks @boazmohar !
